### PR TITLE
feat(new): improve new contract with a ContractType for categorization

### PR DESCRIPTION
**Pull request by peterwht**

### CODE REVIEW

**Comment by peterwht** [comment]
Will refactor `ContractType` and `Provider` types to a reusable type.

File: crates/pop-cli/src/commands/new/contract.rs [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
 #[derive(Args, Clone)]
 pub struct NewContractCommand {
 	#[arg(help = "Name of the contract")]
 	pub(crate) name: Option<String>,
+	#[arg(
+		help = "Contract type.",
+		default_value = ContractType::Examples.as_ref(),
+		value_parser = crate::enum_variants!(ContractType)
+	)]
+	pub(crate) contract_type: Option<ContractType>,
```
Comment (AlexD10S) [inline crates/pop-cli/src/commands/new/contract.rs:24]: I received feedback on a pending PR suggesting that we should make the flags shorter. I like it, do you think makes sense to call it only type? https://github.com/r0gue-io/pop-cli/pull/209#discussion_r1650551879

**Review by AlexD10S — APPROVED** [review]
LGTM! just left a comment

File: crates/pop-cli/src/commands/new/contract.rs (hunk 1 above)
Comment (peterwht) [reply crates/pop-cli/src/commands/new/contract.rs:24]: Unfortunately, type is a reserved keyword in Rust. Could maybe do `category` instead?

Comment (AlexD10S) [reply crates/pop-cli/src/commands/new/contract.rs]: Let's keep it contract_type then, c_type is a bit confusing

**Pull request merged**